test/saithrift/tests/sail3.py

```diff
@@ -2025,3 +2025,431 @@ def runTest(self):
+                                eth_src=router_mac,
+                                ip_dst='10.10.10.1',
+                                ip_src='192.168.0.1',
+                                ip_id=105,
+                                ip_ttl=63)
+
+        pkt2 = simple_ip_packet(pktlen=1600,
+                                eth_dst=router_mac,
+                                eth_src='00:22:22:22:22:22',
+                                ip_dst='10.10.10.1',
+                                ip_src='192.168.0.1',
+                                ip_id=105,
+                                ip_ttl=64)
+
+        exp_pkt2 = simple_ip_packet(pktlen=1600,
+                                eth_dst='00:11:22:33:44:55',
+                                eth_src=router_mac,
+                                ip_dst='10.10.10.1',
+                                ip_src='192.168.0.1',
+                                ip_id=105,
+                                ip_ttl=63)
+        try:
+            send_packet(self, 1, str(pkt1))
+            verify_packet(self, exp_pkt1, 0)
+            # send ip packet to interface 1 to interface 0 (Packet size > MTU)
+            send_packet(self, 1, str(pkt2))
+            verify_no_packet(self, exp_pkt2, 0)
+        finally:
+            sai_thrift_remove_route(self.client, vr_id, addr_family, ip_addr1_subnet, ip_mask1, rif_id1)
+            self.client.sai_thrift_remove_next_hop(nhop1)
+            sai_thrift_remove_neighbor(self.client, addr_family, rif_id1, ip_addr1, dmac1)
+
+            self.client.sai_thrift_remove_router_interface(rif_id1)
+            self.client.sai_thrift_remove_router_interface(rif_id2)
+
+            self.client.sai_thrift_remove_virtual_router(vr_id)
+
+
+@group('l3')
+class L3IPv4NeighborMacTest(sai_base_test.ThriftInterfaceDataPlane):
+    def runTest(self):
+        """
+        Description:
+        Create two VLAN router interfaces in the same VRF. Create nhop, route and
+        neighbor entry for destination. Send packet on one router interface and
+        verify packet received on the other router interface. Simulate a MAC address
+        change for the neighbor entry and verify traffic. The packet must now be
+        forwarded with the new MAC address as destination MAC
+
+        Steps:
+        1. Create VLAN 100, 200 in the database
+        2. Associate 2 tagged ports (port 1, 2) as the member port of VLAN 100 and 200 respectively.
+        3. Create Virtual router V1 and enable V4.
+        4. Create two virtual router interfaces and set the interface type as VLAN
+        5. Create IPv4 neighbor entry with MAC1 and asociate with ""RIF Id1"".
+        6. Create next hop, route to reach the neighbor (MAC1).
+        7. Send test packet from port 1 with src_mac = MAC1 address and dst_mac = router MAC.
+        8. Send traffic on port 2 and observe traffic forwarding to port 1.
+        9. Change the neighbor attribute to update with new MAC (MAC3)
```

Comment:
You can add a step here to send the packet with MAC1 still and see the behavior before learning FDB entry for MAC3.